content/doc/book/pipeline/scaling-pipeline.adoc

```diff
@@ -0,0 +1,88 @@
+
+A *"graceful" shutdown* is where Jenkins goes through a full shutdown process, such as visiting http://[jenkins-server]/exit,  or using normal service shutdown scripts (if Jenkins is healthy).  Sending a SIGTERM/SIGINT to Jenkins will trigger a graceful shutdown.  
+
+A *"dirty" shutdown* is when Jenkins does not get to do normal shutdown processes. This can occur if the process is forcibly terminated.  The most common causes are using SIGKILL to terminate the Jenkins process or killing the container/VM running Jenkins.  Simply stopping or pausing the container/VM will not cause this, as long as the Jenkins process is able to resume. 
+A dirty shutdown can also happen due to catastrophic operating system failures, including the Linux OOMKiller attacking the Jenkins java process to free memory.
+
+All but the most durable setting ("MAX_SURVIVABILITY") currently avoid atomic writes -- what this means is that if the Operating System fails, data that is buffered for writing to disk will not be flushed, it will be lost.  This is quite rare, but can happen as a result of container or virtualization operations that halt the operating system or disconnect storage.  Usually this data is flushed pretty quickly to disk, so the window for data loss is brief.  On Linux this flush-to-disk can be forced by running 'sync'.
+
+== Requirements To Use Durability Settings
+
+* Jenkins LTS 2.73+ or higher (or a weekly 2.62+)
+* Current Pipeline Plugin versions from near the end of January 2018 for all the following:
+    - Pipeline: API (workflow-api)
+    - Pipeline: Groovy (workflow-cps)
+    - Pipeline: Job (workflow-job)
+    - Pipeline: Supporting APIs (workflow-support)
+    - Pipeline: Multibranch (workflow-multibranch) - optional, only needed to enable this setting for multibranch pipelines.
+* Restart the master to use the updated plugins - note: you need all of them to take advantage.
+
+
+== What Are The Durability Settings? 
+
+* Performance optimized mode ("PERFORMANCE_OPTIMIZED") - *Greatly* reduces disk I/O.  If Pipelines do not finish AND Jenkins is not shut down gracefully, they may lose data and behave like freestyle projects -- see details above.
+
+* Maximum durability ("MAX_SURVIVABILITY") - behaves just like Pipeline did before, slowest option.  Use this for running your most critical Pipelines.
+
+* Less durable, a bit faster ("SURVIVABLE_NONATOMIC") - Writes data with every step but avoids atomic writes. This is faster than maximum durability mode, especially on networked filesystems.  It carries a small extra risk (details above under "What Am I Giving Up").
```

Comment:
Details are unspecific about what circumstances this still protects from, and which not.